tests/test_cli.py

```diff
@@ -54,30 +76,25 @@ class TestToolCLI(unittest.TestCase):
     @classmethod
     def tearDownClass(cls):
         for tempdir in cls.temporary_directories:
-            tempdir.cleanup()
+            rmtree(tempdir.name, ignore_errors=True)
         for tempfile in cls.temporary_files:
             os.unlink(tempfile)
 
     def check_no_args(self, context):
         """Check running the tool with no arguments."""
-        # if the BINPATH environment variable is defined
-        # use that to construct the command instead of just
-        # the name; this is needed for the CI builds where
-        # we do not always activate the conda environment
-        binpath = os.environ.get("BINPATH", None)
-        if binpath is not None:
-            cmd = f"{binpath}/{context}"
-        else:
-            cmd = f"{context}"
+        # call the main function for the given context
+        # with no arguments and check that it exits
+        # with exit code 0 and prints out the usage
+        main_function = MAIN_FUNCTIONS[context]
+        with redirect_stdout(StringIO()) as captured_stdout:
+            with self.assertRaises(SystemExit) as excm:
+                main_function([])
```

Comment:
shouldn't we test it with no params, like that: `main_function()`, then argue is None and it will cover the lines above with the warnings?

Reply:
Yes,  I tried that but the problem is if `argv` is `None`, it relies on `sys.argv` which is then going to be the `nose2` arguments 😉.

Reply:
This test case does test what'd happen if we just run `rsmtool` without any arguments since the `rsmtool` entry point points to `rsmtool.rsmtool.main`.

Reply:
but will it be `None`, or `[]` if we run it with no arguments? If it's an empty list, when will it be `None`?

Reply:
Yes, it will be `None` but since `sys.argv[1:]` will be an empty list, it's the same as passing `[]`.  See first two lines in `rsmtool.main()`.

Reply:
oh, got it now!